kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/KafkaUtils.java

```diff
@@ -0,0 +1,208 @@
+        return pods.list().getItems().get(pods.list().getItems().size() - 1).getMetadata().getName();
+    }
+
+    /**
+     * Produce message.
+     *
+     * @param deployNamespace the deploy namespace
+     * @param topicName the topic name
+     * @param message the message
+     * @param bootstrap the bootstrap
+     */
+    public static void ProduceMessage(String deployNamespace, String topicName, String message, String bootstrap) {
+        kubeClient().getClient().run().inNamespace(deployNamespace).withNewRunConfig()
+                .withImage(Constants.STRIMZI_KAFKA_IMAGE)
+                .withName("java-kafka-producer")
+                .withRestartPolicy("Never")
+                .withCommand("/bin/sh")
+                .withArgs("-c", "echo '" + message + "'| bin/kafka-console-producer.sh --bootstrap-server " + bootstrap + " --topic " + topicName)
+                .done();
+    }
+
+    /**
+     * Produce message with test clients.
+     *
+     * @param deployNamespace the deploy namespace
+     * @param topicName the topic name
+     * @param bootstrap the bootstrap
+     * @param message the message
+     * @param numOfMessages the num of messages
+     * @return the string
+     * @throws IOException the io exception
+     */
+    public static String produceMessageWithTestClients(String deployNamespace, String topicName, String bootstrap, String message, int numOfMessages) throws IOException {
+        LOGGER.debug("Producing {} messages in '{}' topic", numOfMessages, topicName);
+        File file = replaceStringInResourceFile("kafka-producer-template.yaml", "kafka-producer.yaml", Map.of(
+                "%BOOTSTRAP_SERVERS%", bootstrap,
+                "%TOPIC_NAME%", topicName,
+                "%MESSAGE_COUNT%", "\"" + numOfMessages + "\"",
+                "%MESSAGE%", message));
+        kubeClient().getClient().load(new FileInputStream(file)).inNamespace(deployNamespace).create();
+        return getPodNameByLabel(deployNamespace, "app", Constants.KAFKA_PRODUCER_CLIENT_LABEL, Duration.ofSeconds(10).toMillis());
+    }
+
+    private static File replaceStringInResourceFile(String resourceTemplateFileName, String newResourceFileName, Map<String, String> replacements) throws IOException {
+        Path path = Path.of(Objects.requireNonNull(KafkaUtils.class
+                .getClassLoader().getResource(resourceTemplateFileName)).getPath());
+        Charset charset = StandardCharsets.UTF_8;
+
+        String content = new String(Files.readAllBytes(path), charset);
+        for (Map.Entry<String, String> entry : replacements.entrySet()) {
+            content = content.replaceAll(entry.getKey(), entry.getValue());
+        }
+        String resourceFile = "/tmp/" + newResourceFileName;
+
+        Files.writeString(Path.of(resourceFile), content, charset);
+        File file = new File(resourceFile);
+        file.deleteOnExit();
+        return file;
```

Comment:
You probably don't actually need to create an temporary file.  You can stream from memory.

InputStream targetStream = new ByteArrayInputStream(content.getBytes());
return targetStream;

Reply:
done